File: RedSheet.DbEntities/Models/PowerReportInformation.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("PowerReportInformations", Schema = "dbo")]
	[Serializable]
	public partial class PowerReportInformation  
	{
 
	[MaxLength(100)]
	public string ImageName { get; set; } 
 
	[Required]
	public bool OurPowerKnowledge { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int PowerReportInformationId { get; set; } 
 
	[Required]
	public bool TheirPowerKnowledge { get; set; } 
 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	//Relationships
	public int ActualPowerId { get; set; } 

	public ApplicationObject ApplicationObject { get; set; } 
 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	//Relationships
	public int ProjectedPowerId { get; set; } 

	public ApplicationObject ApplicationObject1 { get; set; } 

	public PowerReportInformation()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/power-report-information.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ApplicationObject,  } from './'
export class PowerReportInformation {
    constructor(powerReportInformation?: PowerReportInformation )  {
        let properties = [ "imageName", "ourPowerKnowledge", "powerReportInformationId", "theirPowerKnowledge", "actualPowerId", "projectedPowerId",];
        for (let property of properties)
            if (powerReportInformation && powerReportInformation[property])
                this[property] = powerReportInformation[property];
    }
 
    @maxLength(100)
	imageName : string =   undefined;
 
	ourPowerKnowledge : boolean = false ;
 
	powerReportInformationId : number =   0 ;
 
	theirPowerKnowledge : boolean = false ;
 
    @range(0,2147483647)
	actualPowerId : number =   undefined;
	applicationObject : ApplicationObject  ;
 
    @range(0,2147483647)
	projectedPowerId : number =   undefined;
	applicationObject1 : ApplicationObject  ;


}
